Sources/PointFree/Subscribe.swift

```diff
@@ -150,15 +196,26 @@ private func validateQuantity(_ pricing: Pricing) -> Bool {
 
 private func loginAndRedirectToPricing<A>(
   _ conn: Conn<StatusLineOpen, A>
-  )
+)
   -> IO<Conn<ResponseEnded, Data>> {
 
-  return conn
-    |> redirect(to: .login(redirect: url(to: .pricingLanding)))
+    return conn
+      |> redirect(to: .login(redirect: url(to: .pricingLanding)))
 }
 
 private func validateCoupon(forSubscribeData subscribeData: SubscribeData) -> Bool {
-  return subscribeData.coupon == nil || subscribeData.pricing.quantity == 1
+  subscribeData.coupon == nil
+    // Do not allow using coupons on team subscriptions
+    || subscribeData.pricing.quantity == 1
+    // Do not allow using regional discount coupon id directly
+    && subscribeData.coupon != Current.envVars.regionalDiscountCouponId
```

Comment:
making sure that if the regional discount coupon id leaks that people can't use it except through that dedicated flow.